code/modules/mining/machine_processing.dm

```diff
@@ -82,39 +82,42 @@
 	else
 		return INITIALIZE_HINT_QDEL
 
-/obj/machinery/mineral/processing_unit_console/ui_interact(mob/user)
-	. = ..()
-	if(!processing_machine)
-		return
+/obj/machinery/mineral/processing_unit_console/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ProcessingConsole")
+		ui.open()
 
-	var/dat = processing_machine.get_machine_data()
+/obj/machinery/mineral/processing_unit_console/ui_static_data(mob/user)
+	return processing_machine.ui_static_data()
 
-	var/datum/browser/popup = new(user, "processing", "Smelting Console", 300, 500)
-	popup.set_content(dat)
-	popup.open()
+/obj/machinery/mineral/processing_unit_console/ui_data(mob/user)
+	return processing_machine.ui_data()
 
-/obj/machinery/mineral/processing_unit_console/Topic(href, href_list)
-	if(..())
+/obj/machinery/mineral/processing_unit_console/ui_act(action, list/params)
+	. = ..()
+	if(.)
 		return
-	usr.set_machine(src)
-	add_fingerprint(usr)
 
-	if(href_list["material"])
-		var/datum/material/new_material = locate(href_list["material"])
-		if(istype(new_material))
+	switch(action)
+		if("setMaterial")
+			var/datum/material/new_material = locate(params["value"])
+			if(!istype(new_material))
+				return
+
 			processing_machine.selected_material = new_material
 			processing_machine.selected_alloy = null
+			return TRUE
 
-	if(href_list["alloy"])
-		processing_machine.selected_material = null
-		processing_machine.selected_alloy = href_list["alloy"]
-
-	if(href_list["set_on"])
-		processing_machine.on = (href_list["set_on"] == "on")
-		processing_machine.begin_processing()
+		if("setAlloy")
+			processing_machine.selected_material = null
+			processing_machine.selected_alloy = params["value"]
+			return TRUE
 
-	updateUsrDialog()
-	return
+		if("toggle")
+			processing_machine.on = !processing_machine.on
+			processing_machine.begin_processing()
```

Comment:
```suggestion
                      if(on)
			  processing_machine.begin_processing()
		     else
		         processing_machine.end_processing()
```

Reply:
Actually, what's the point of doing this when this part of logic is handled inside `Process` proc?